cluster-autoscaler/cloudprovider/gce/templates_test.go

```diff
@@ -322,6 +326,96 @@ func TestBuildAllocatableFromKubeEnv(t *testing.T) {
 		}
 	}
 }
+func TestBlockedBootDiskForEphemeralStorage(t *testing.T) {
```

Comment:
This feels like it's almost a copy of TestBuildNodeFromTemplateSetsResources, which makes sense given that you're testing an condition checked in BuildNodeFromTemplate that is related to setting resources. At a glance it looks like maybe a better way to do it would be to add isEpemeralStorageBlocked and the related if statement to TestBuildNodeFromTemplateSetsResources and just add cases related to this logic there?

Reply:
Done. Removed it.
I just have a concern, should we look to not add more cases to a single test func TestBuildNodeFromTemplateSetsResources, rather have smaller test functions that test specific functionalities.

Reply:
In general I agree - we should aim to have multiple unittests covering separate pieces of code in isolation. However, in this specific case the test was almost an exact copy case of the other test and vast majority of logic executed by test was the same. I don't see the benefit of artificially splitting very similar tests and I think it makes it harder for future maintainers who will need to read ~the same thing twice and spend time figuring what the differences are.